lib/assert.js

```diff
@@ -341,6 +357,191 @@ assert.notStrictEqual = function notStrictEqual(actual, expected, message) {
+    }
+    if (typeof actual === 'string') {
+      return StringPrototypeIncludes(actual, expected);
+    }
+
+    return ArrayPrototypeEvery(expected, (item) => ArrayPrototypeSome(actual, (actualItem) => {
+      // eslint-disable-next-line eqeqeq
+      return loose ? actualItem == item : actualItem === item;
+    }));
+  }
+
+  // Check non object types equality
+  if (!isPlainObject(actual) || !isPlainObject(expected)) {
+    if (isDeepEqual === undefined) {
+      lazyLoadComparison();
+    }
+    return loose ? isDeepEqual(actual, expected) : isDeepStrictEqual(actual, expected);
+  }
+
+  // Use Reflect.ownKeys() instead of Object.keys() to include symbol properties
+  const keysExpected = ReflectOwnKeys(expected);
+
+  // Handle circular references
+  if (SetPrototypeHas(comparedObjects, actual)) {
+    return true;
+  }
+  comparedObjects.add(actual);
+
+  // Check if all expected keys and values match
+  for (let i = 0; i < keysExpected.length; i++) {
+    const key = keysExpected[i];
+    assert(
+      ReflectHas(actual, key),
+      new AssertionError({ message: `Expected key ${String(key)} not found in actual object` }),
+    );
+    if (!compareBranch(actual[key], expected[key], loose, comparedObjects)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/**
+ * The strict equivalence assertion test between two objects
+ * @param {any} actual
+ * @param {any} expected
+ * @param {string | Error} [message]
+ * @returns {void}
+ */
+assert.partialDeepStrictEqual = function partialDeepStrictEqual(
+  actual,
+  expected,
+  message,
+) {
+  if (arguments.length < 2) {
+    throw new ERR_MISSING_ARGS('actual', 'expected');
+  }
+
+  if (!compareBranch(actual, expected)) {
+    innerFail({
+      actual,
+      expected,
+      message,
+      operator: 'partialDeepStrictEqual',
+      stackStartFn: partialDeepStrictEqual,
+    });
+  }
+};
+
+/**
+ * The equivalence assertion test between two objects
+ * @param {any} actual
+ * @param {any} expected
+ * @param {string | Error} [message]
+ * @returns {void}
+ */
+assert.partialDeepEqual = function partialDeepEqual(actual, expected, message) {
+  if (arguments.length < 2) {
+    throw new ERR_MISSING_ARGS('actual', 'expected');
+  }
+
+  if (!compareBranch(actual, expected, true)) {
+    innerFail({
+      actual,
+      expected,
+      message,
+      operator: 'partialDeepEqual',
+      stackStartFn: partialDeepEqual,
+    });
+  }
+};
+
+/**
+ * The inclusion assertion test between two arrays or strings
+ * @param {Array | string} actual
+ * @param {any} expected
+ * @param {string | Error} [message]
+ * @returns {void}
+ */
+assert.includes = function includes(actual, expected, message) {
+  if (arguments.length < 2) {
+    throw new ERR_MISSING_ARGS('actual', 'expected');
+  }
+
+  if (typeof actual !== 'string' && !ArrayIsArray(actual)) {
+    throw new ERR_INVALID_ARG_TYPE('actual', ['string', 'Array'], actual);
+  }
+
+  if (!actual.includes(expected)) {
```

Comment:
```suggestion
  const isString = typeof actual === 'string';

  if (!isString && !ArrayIsArray(actual)) {
    throw new ERR_INVALID_ARG_TYPE('actual', ['string', 'Array'], actual);
  }

  if (!(isString ? StringPrototypeIncludes : ArrayPrototypeIncludes)(actual, expected)) {
```

Can we add a test that includes a null-prototype array (`Object.setPrototypeOf([], null)`) to ensure `assert.includes` still works in this case?